### ISSUE TITLE
Update Overview page for Domain management

### ISSUE THREAD

**Issue by gabydisarli** [cisagov/manage.get.gov#2460]
### Issue description

As part of the Org model design work, some small changes have been made to the Domain Overview page to accomodate requirements. We should make these changes across manage even for non-org users. The changes are:

1. Expose whether a Domain manager has enabled DNSSEC on the Overview page for a domain. This change should happen across the registrar, but it's especially key for the Org model designs as not all users will have access to the pages within a domain, but instead only be able to see the domain's Overview page. 
2. Changing the heading structure to the Domain overview to match styling across other pages. 

### Acceptance criteria

[Figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=4788-13705&t=iP1kVCD82km3Hm6p-11) for reference
- [ ] Add new section to Overview page following 
- [ ] The text under DNSSEC says either "Enabled" or "Not enabled"
- [ ] Change page title (h1) to "Domain overview"
- [ ] Add name of domain as h2 above sections
- [ ] Change section headings to h3.

### Current overview page
![Screenshot 2024-07-12 at 3 44 14 PM](https://github.com/user-attachments/assets/db363735-4b0c-4786-af8f-ede8ba86230a)

###  New Overview page

![Screenshot 2024-07-12 at 3 56 32 PM](https://github.com/user-attachments/assets/c18af621-48ab-4f97-8e19-746f25c609f3)

**Comment by h-m-f-t** [comment]
I like this but think the positioning of the status, expires, and date created fields is awkward in the new view because they're more relevant to the new h2 instead of the new h1. Do you think we should keep them on top, @gabydisarli?

**Comment by gabydisarli** [comment]
> I like this but think the positioning of the status, expires, and date created fields is awkward in the new view because they're more relevant to the new h2 instead of the new h1. Do you think we should keep them on top, @gabydisarli?

Good catch! I think it makes sense to have h2 above status, expires, and creation date. I'll update Figma and this ticket.

**Comment by gabydisarli** [comment]
The Figma and ticket have been updated

**Comment by abroddrick** [comment]
tentatively added to next sprint, will evaluate after refinement on monday if it's good to go.

**Comment by abroddrick** [comment]
I think this one is fine, but let's talk in refinement to make sure nothing is missed. I moved it to ready to indicate it looks good to me!
